cpp/oneapi/dal/algo/jaccard/vertex_similarity_types.cpp

```diff
@@ -15,48 +15,53 @@
  *******************************************************************************/
 
 #include "oneapi/dal/algo/jaccard/vertex_similarity_types.hpp"
-#include <iostream>
 #include "oneapi/dal/algo/jaccard/common.hpp"
 
 namespace oneapi::dal::preview {
 namespace jaccard {
+template <typename Graph>
 class detail::similarity_input_impl : public base {
 public:
-    similarity_input_impl(const graph &graph_data_input) : graph_data(graph_data_input) {}
+    similarity_input_impl(const Graph &graph_data_input) : graph_data(graph_data_input) {}
 
-    const graph &graph_data;
+    const Graph &graph_data;
 };
 
 using detail::similarity_input_impl;
 
-similarity_input::similarity_input(const graph &data) : impl_(new similarity_input_impl(data)) {}
+template <typename Graph>
+similarity_input<Graph>::similarity_input(const Graph &data)
+        : impl_(new similarity_input_impl<Graph>(data)) {}
 
-const graph &similarity_input::get_graph() const {
+template <typename Graph>
+const Graph &similarity_input<Graph>::get_graph() const {
     return impl_->graph_data;
 }
 
 class detail::similarity_result_impl : public base {
 public:
-    similarity_result_impl(const array_of_floats &coefficients,
-                           const array_of_pairs_uint32t &vertex_pairs)
-            : coefficients(coefficients),
+    similarity_result_impl(const table &coeffs, const table &vertex_pairs)
+            : coeffs(coeffs),
               vertex_pairs(vertex_pairs) {}
 
-    array_of_floats coefficients;
-    array_of_pairs_uint32t vertex_pairs;
+    table coeffs;
+    table vertex_pairs;
 };
 
+template class detail::similarity_input_impl<undirected_adjacency_array_graph<> &>;
+
+template class similarity_input<undirected_adjacency_array_graph<> &>;
+
 using detail::similarity_result_impl;
 
-similarity_result::similarity_result(const array_of_floats &coefficients,
-                                     const array_of_pairs_uint32t &vertex_pairs)
-        : impl_(new similarity_result_impl(coefficients, vertex_pairs)) {}
+similarity_result::similarity_result(const table &coeffs, const table &vertex_pairs)
+        : impl_(new similarity_result_impl(coeffs, vertex_pairs)) {}
 
-array_of_floats similarity_result::get_jaccard_coefficients() const {
-    return impl_->coefficients;
+table similarity_result::get_coeffs() const {
+    return impl_->coeffs;
 }
 
-array_of_pairs_uint32t similarity_result::get_vertex_pairs() const {
+table similarity_result::get_vertex_pairs() const {
```

Comment:
is this table, const table, or const table&?
How does ml structures signatures in such cases?

Reply:
@andreynv 
they  return table:
[table get_eigenvectors() const;](https://github.com/bysheva/oneDAL/blob/artemaro/cpp/oneapi/dal/algo/pca/train_types.hpp)

Reply:
const table& should be returned type, dal would align codes with the specs https://spec.oneapi.com/versions/latest/elements/oneDAL/source/algorithms/decomposition/pca.html